traffic_router/ultimate-test-harness/http_test.go

```diff
@@ -0,0 +1,497 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"text/tabwriter"
+	"time"
+
+	go_log "github.com/apache/trafficcontrol/lib/go-log"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	client "github.com/apache/trafficcontrol/traffic_ops/v4-client"
+
+	"github.com/kelseyhightower/envconfig"
+)
+
+const (
+	UserAgent = "Traffic Router Load Tests"
+)
+
+type TOConfig struct {
+	TOURL      string `required:"true" envconfig:"TO_URL"`
+	TOUser     string `required:"true" envconfig:"TO_USER"`
+	TOPassword string `required:"true" envconfig:"TO_PASSWORD"`
+	TOInsecure bool   `default:"true"  envconfig:"TO_INSECURE"`
+	TOTimeout  int    `default:"30"    envconfig:"TO_TIMEOUT"`
+}
+
+type TRDetails struct {
+	Hostname           string
+	IPAddresses        []string
+	ClientIP           string
+	ClientIPAddressMap IPAddressMap
+	Port               int
+	DSHost             string
+	CDNName            tc.CDNName
+}
+
+type IPAddressMap struct {
+	Zones []string
+	Map   map[string]tc.CoverageZoneLocation
+}
+
+type Benchmark struct {
+	RequestsPerSecondThreshold int
+	BenchmarkTime              int
```

Comment:
Since this is in seconds, can we call it `BenchmarkSeconds`?

Reply:
Renamed to `BenchmarkSeconds` in 4fd61ae4b9